packages/babel-helper-create-class-features-plugin/src/decorators.ts

```diff
@@ -277,6 +278,84 @@ function extractProxyAccessorsFor(
   ];
 }
 
+function getComputedKeyCompletion(path: NodePath<t.Expression>) {
+  if (path.isSequenceExpression()) {
+    const expressions = path.get("expressions");
+    return getComputedKeyCompletion(expressions[expressions.length - 1]);
+  }
+  return skipTransparentExprWrappers(path);
+}
+
+function getComputedKeyMemoiser(path: NodePath<t.Expression>): t.Expression {
+  const completion = getComputedKeyCompletion(path);
+  if (completion.isConstantExpression()) {
+    return t.cloneNode(path.node);
+  } else if (
+    completion.isAssignmentExpression() &&
+    completion.get("left").isIdentifier()
+  ) {
+    return t.cloneNode(completion.node.left);
+  } else {
+    throw new Error(
+      "Internal Error: the computed key has not yet been memoised.",
+    );
+  }
+}
+
+function prependExpressionsToComputedKey(
+  expressions: t.Expression[],
+  fieldPath: NodePath<
+    t.ClassMethod | t.ClassProperty | t.ClassAccessorProperty
+  >,
+) {
+  const key = fieldPath.get("key") as NodePath<t.Expression>;
+  expressions.push(key.node);
```

Comment:
If a `key` is already a sequence expression, it could be generated by another Babel plugin. The intention here is try to reuse the uid memoiser, for example the computed keys transform can reuse the memoiser `_computedKeys` generated by the decorator transform:

```js
class C {
  [decs = f(), _computedKey = g(), another_decs = h(), _computedKey];
}
```

but at this point there is no class features running before decorators, so there aren't much practical examples. I will add a test case for nested sequence expression.

Reply:
6969fd41db1aef76a8c0e63e5551ee2bb025d592